oap-server/server-storage-plugin/storage-elasticsearch-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/elasticsearch/base/TimeSeriesUtils.java

```diff
@@ -60,6 +61,31 @@ public static String latestWriteIndexName(Model model) {
         }
     }
 
+    /**
+     * @return Concrete index name for super dataset index
+     */
+    public static String[] querySuperDatasetIndices(String indexName, long startTimeBucket, long endTimeBucket) {
+        if (startTimeBucket == 0 || endTimeBucket == 0) {
+            return new String[] {indexName};
+        }
+        long startDay = compressTimeBucket(convert2DayTimeBucket(startTimeBucket), SUPER_DATASET_DAY_STEP);
+        long endDay = compressTimeBucket(convert2DayTimeBucket(endTimeBucket), SUPER_DATASET_DAY_STEP);
+        DateTime startDateTime = TIME_BUCKET_FORMATTER.parseDateTime(startDay + "");
+        DateTime endDateTime = TIME_BUCKET_FORMATTER.parseDateTime(endDay + "");
+
+        int steps = Math.max((Days.daysBetween(startDateTime, endDateTime).getDays()) / SUPER_DATASET_DAY_STEP, 0);
```

Comment:
Max with `0`? Do you avoid negative value? What happens when `the steps == 0`?

Reply:
I think this method should simply follow this logic,

1. List all days between `startTime` and `endTime`.
1. Use `compressTimeBucket` to get all indexes(could have duplicated records)
1. Distinct the list.

Reply:
The above processes may need a little refactor about the `compressTimeBucket` method, I should be able to accept `DateTime` as one parameter too.